File: scalademo/src/com/milko/trainings/scala/interfaces/scalac/FileReader.scala
package com.milko.trainings.scala.interfaces.scalac

import java.io.File
import java.nio.CharBuffer

class FileReader(file:File) extends Readable with AutoCloseable{
  override def read(buffer:CharBuffer):Int = {
    val linesRead: Int = 0;
    linesRead;
  }
  
  override def close:Unit = ???
}

File: scalademo/src/com/milko/trainings/scala/interfaces/javac/FileReader.java
package com.milko.trainings.scala.interfaces.javac;

import java.io.File;
import java.nio.CharBuffer;

public class FileReader implements Readable, AutoCloseable {
	
	private File file;
	
	public FileReader(File file) {
		this.file = file;
	}

	@Override
	public int read(CharBuffer buffer) {
		int read = 0;
		
		return 0;
	}

	@Override
	public void close() throws Exception {
		//close
		
	}

}
